pkg/cmdline/cmdline.go

```diff
@@ -29,54 +25,29 @@ type CmdLine struct {
 	Err   error
 }
 
-var (
-	// procCmdLine package level static variable initialized once
-	once        sync.Once
-	procCmdLine CmdLine
-)
-
-func cmdLineOpener() {
-	cmdlineReader, err := os.Open("/proc/cmdline")
-	if err != nil {
-		errorMsg := fmt.Sprintf("Can't open /proc/cmdline: %v", err)
-		log.Print(errorMsg)
-		procCmdLine = CmdLine{Err: fmt.Errorf(errorMsg)}
-		return
-	}
-
-	procCmdLine = parse(cmdlineReader)
-	cmdlineReader.Close()
-}
-
 // NewCmdLine returns a populated CmdLine struct
-func NewCmdLine() CmdLine {
-	// We use cmdLineReader so tests can inject here
-	once.Do(cmdLineOpener)
-	return procCmdLine
+func NewCmdLine() *CmdLine {
+	return getCmdLine()
 }
 
 // FullCmdLine returns the full, raw cmdline string
 func FullCmdLine() string {
-	once.Do(cmdLineOpener)
-	return procCmdLine.Raw
+	return getCmdLine().Raw
 }
 
 // parse returns the current command line, trimmed
-func parse(cmdlineReader io.Reader) CmdLine {
+func parse(cmdlineReader io.Reader) *CmdLine {
+	var line CmdLine
 	raw, err := io.ReadAll(cmdlineReader)
-	line := CmdLine{}
-	if err != nil {
-		log.Printf("Can't read command line: %v", err)
+	if line.Err != nil {
```

Comment:
should this be `if err != nil` ?   line.Err isn't set until the line below, line 43, so it can't be checked here?

Reply:
yes, sorry.